docs/DEPENDENCIES.md

```diff
@@ -0,0 +1,306 @@
+
+---
+
+### Quality of Life Dependencies
+
+#### CLI Framework
+
+**Option 1: github.com/urfave/cli/v2 (Recommended)**
+```go
+github.com/urfave/cli/v2 v2.25.7
+```
+
+**Pros:**
+- Simpler implementation than cobra
+- Lighter weight dependency
+- Good subcommand support
+- Clean API for basic commands
+
+**Cons:**
+- Smaller ecosystem than cobra
+
+**Option 2: github.com/spf13/cobra**
+```go
+github.com/spf13/cobra v1.7.0
+```
+
+**Pros:**
+- Industry standard (used by kubectl, docker, etc.)
+- Excellent subcommand support
+- Built-in help generation
+
+**Cons:**
+- More setup boilerplate
+- Larger dependency footprint
+- Over-engineered for 4 simple commands
+
+**Option 3: Standard Library flag**
+```go
+// No external dependency
+import "flag"
+```
+
+**Pros:**
+- Zero dependencies
+- Sufficient for basic CLI needs
+
+**Cons:**
+- Manual subcommand handling
+- More boilerplate code
+- Less polished UX
+
+**Decision: Use urfave/cli/v2**
+- Simpler implementation for basic subcommands
+- Less boilerplate than cobra
+- Adequate CLI UX - users just need it to work
+- Lighter dependency footprint
+
+---
+
+#### Logging
+
+**Option 1: slog (Recommended)**
+```go
+// Go 1.21+ standard library
+import "log/slog"
+```
+
+**Pros:**
+- Part of standard library (Go 1.21+)
+- Structured logging support
+- Good performance
+- Modern API design
+
+**Cons:**
+- Requires Go 1.21+
+
+**Option 2: github.com/sirupsen/logrus**
+```go
+github.com/sirupsen/logrus v1.9.3
+```
+
+**Pros:**
+- Battle-tested
+- Rich ecosystem
+- Works with older Go versions
+
+**Cons:**
+- External dependency
+- Maintenance mode (frozen API)
+
+**Decision: Use slog**
+- Standard library preference
+- Modern structured logging
+- Project targets Go 1.21+ anyway
+
+---
+
+### Optional Dependencies
+
+#### UUID Generation
+
+**Option 1: github.com/google/uuid**
+```go
+github.com/google/uuid v1.4.0
+```
+
+**Pros:**
+- Standard UUID implementation
+- RFC 4122 compliant
+- Battle-tested
+
+**Cons:**
+- Additional dependency for simple functionality
+
+**Option 2: crypto/rand + hex encoding**
+```go
+// Standard library only
+import (
+    "crypto/rand"
+    "encoding/hex"
+)
+
+func generateJobID() string {
+    bytes := make([]byte, 16)
+    rand.Read(bytes)
+    return hex.EncodeToString(bytes)
+}
+```
+
+**Pros:**
+- No external dependencies
+- Cryptographically secure
+- Sufficient randomness for demo
+
+**Cons:**
+- Not RFC 4122 compliant
+- Manual implementation
+
+**Decision: Use crypto/rand approach**
+- Avoids dependency for simple use case
+- Demonstrates standard library preference
+- Adequate for demonstration purposes
+
+---
+
+#### Testing Utilities
+
+**Option 1: Standard Library testing**
+```go
+import "testing"
+```
+
+**Pros:**
+- No dependencies
+- Sufficient for most testing needs
+- Encourages simple test design
+
+**Cons:**
+- More verbose assertions
+- No test fixtures/helpers
+
+**Option 2: github.com/stretchr/testify**
```

Comment:
We're fans of testify, feel free to use it in your tests to simplify the assertions.